cmd/config.go

```diff
@@ -274,12 +276,29 @@ type BeelineConfig struct {
 	Dataset string
 	// SampleRate is the (positive integer) denominator of the sample rate.
 	// Default: 1 (meaning all traces are sent). Set higher to send fewer traces.
-	SampleRate uint
+	SampleRate uint32
 	// Mute disables honeycomb entirely; useful in test environments.
 	Mute bool
 	// Many other fields of beeline.Config are omitted as they are not yet used.
 }
 
+// makeSampler constructs a SamplerHook which will deterministically decide if
+// any given span should be sampled based on its TraceID, which is shared by all
+// spans within a trace.
+func makeSampler(rate uint32) func(fields map[string]interface{}) (bool, int) {
+	if rate == 0 {
```

Comment:
nit: I would prefer if the zero value of this arg wasn't mutated to `1`. In the `beeline-go` package they return an error if you pass `0` instead.

Reply:
Yes, but that sampler is optional. Here, the sampler is being non-optionally constructed from a SampleRate, and by default a SampleRate of 0 is equivalent to a SampleRate of 1:
https://github.com/honeycombio/beeline-go/blob/2df51661f57e5ed87f43ee62cc3993c3f78ec4d9/beeline.go#L39-L42

I've updated the function comment to document this.

Reply:
Thanks for helping me understand that 😄